### PR TITLE
Fix: ConversationListCellTests without subclassing CoreDataSnapshotTestCase

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Remove `CoreDataSnapshotTestCase` subclassing from `ConversationListCellTests`

### Discussion
- The original tests use snapshot tests to check `ZMConversation` operations results, is there any tests already or we should add some tests for `ZMConversation` operations to generate `ConversationStatus`?

### CODE REVIEW

File: Wire-iOS Tests/ConversationAvatarViewModeTests.swift [hunk 1]
```diff
@@ -18,19 +18,40 @@
 import XCTest
 @testable import Wire
 
-final class MockConversationAvatarViewConversation: SwiftMockConversation, StableRandomParticipantsProvider {
+class MockStableRandomParticipantsConversation: SwiftMockConversation, StableRandomParticipantsProvider {
     var stableRandomParticipants: [UserType] = []
+    
+    required override init() {
```
Comment (typfel) [inline Wire-iOS Tests/ConversationAvatarViewModeTests.swift:24]: Can be removed

**Review by typfel — APPROVED** [review]

File: Wire-iOS Tests/ConversationAvatarViewModeTests.swift (hunk 1 above)
Comment (billypchan) [reply Wire-iOS Tests/ConversationAvatarViewModeTests.swift:24]: It is required to fix the compiler error `Constructing an object of class type 'T' with a metatype value must use a 'required' initializer`

File: Wire-iOS Tests/ConversationList/ConversationListCell/ConversationListCellTests.swift [hunk 2]
```diff
@@ -20,292 +20,480 @@ import UIKit
...
         // when & then
         verify(otherUserConversation)
     }
-    
+
     func testThatItRendersMutedConversation() {
         // when
-        otherUserConversation.mutedMessageTypes = [.all]
-        
+        let status = ConversationStatus(isGroup: false, hasMessages: false, hasUnsentMessages: false, messagesRequiringAttention: [], messagesRequiringAttentionByType: [:], isTyping: false, mutedMessageTypes: [.all], isOngoingCall: false, isBlocked: false, isSelfAnActiveMember: true, hasSelfMention: false, hasSelfReply: false)
```
Comment (johnxnguyen) [inline Wire-iOS Tests/ConversationList/ConversationListCell/ConversationListCellTests.swift:123]: Would be nice to have consistent indentation (like the other tests)

File: Wire-iOS Tests/ConversationList/ConversationListCell/ConversationListCellTests.swift [hunk 3]
```diff
@@ -20,292 +20,480 @@ import UIKit
...
         verify(otherUserConversation)
     }
 
     func testThatItRendersBlockedConversation() {
         // when
-        otherUserConversation.connectedUser?.toggleBlocked()
-        
+        otherUserConversation.connectedUserType?.toggleBlocked()
+
+        let status = ConversationStatus(isGroup: false, hasMessages: false, hasUnsentMessages: false, messagesRequiringAttention: [], messagesRequiringAttentionByType: [:], isTyping: false, mutedMessageTypes: [], isOngoingCall: false, isBlocked: true, isSelfAnActiveMember: true, hasSelfMention: false, hasSelfReply: false)
```
Comment (johnxnguyen) [inline Wire-iOS Tests/ConversationList/ConversationListCell/ConversationListCellTests.swift:134]: Would be nice to have consistent indentation (like the other tests)

File: Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListCell+ZMConversationObserver.swift [hunk 4]
```diff
@@ -20,8 +20,8 @@ import Foundation
 import WireDataModel
 
 extension ConversationListCell: ZMConversationObserver {
-    public func conversationDidChange(_ change: ConversationChangeInfo) {
-        guard change.conversation == self.conversation,
+    func conversationDidChange(_ change: ConversationChangeInfo) {
+        guard change.conversation == (conversation as? ZMConversation),
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListCell+ZMConversationObserver.swift:24]: Maybe a referential equality is more suitable here?

**Review by johnxnguyen — APPROVED** [review]

**Comment by billypchan** [comment]
@johnxnguyen updated for you comments.

**Review by marcomaddy1982 — APPROVED** [review]

File: Wire-iOS Tests/ConversationAvatarViewModeTests.swift (hunk 1 above)
Comment (typfel) [reply Wire-iOS Tests/ConversationAvatarViewModeTests.swift:24]: OK

**Pull request merged**